docker-app/qfieldcloud/subscription/models.py

```
@@ -819,70 +841,45 @@ def create_subscription(
+            "trial_ends_at": None,
+        }
+
+        if trial_plan:
+            trial_end_date = current_time + timedelta(days=config.TRIAL_PERIOD_DAYS)
+
+            subscription_data.update(
+                {
+                    "trial_plan": trial_plan,
+                    "trial_ends_at": trial_end_date,
+                }
             )
-            # NOTE to get annotations, mostly `is_active`
-            trial_subscription_obj = cls.objects.get(pk=trial_subscription.pk)
 
             if (
                 account.user.is_organization
                 and account.user.organization_owner.remaining_trial_organizations > 0
             ):
-                account.user.organization_owner.remaining_trial_organizations -= 1
-                account.user.organization_owner.save(
-                    update_fields=["remaining_trial_organizations"]
-                )
-
-            # the trial plan should be the default plan
-            regular_plan = Plan.objects.get(
-                user_type=account.user.type,
-                is_default=True,
-            )
-
-            # the end date of the trial is the start date of the regular
-            regular_active_since = active_until
-        else:
-            trial_subscription_obj = None
-            regular_plan = plan
-            regular_active_since = active_since
-
-            # NOTE in case the user had a custom amount set (e.g manually set by support) this will
-            # be overwritten by a subscription plan change.
-            # But taking care of this would add quite some complexity.
-            if account.user.is_person:
-                account.user.remaining_trial_organizations = (
-                    regular_plan.max_trial_organizations
-                )
+                account.user.remaining_trial_organizations -= 1
                 account.user.save(update_fields=["remaining_trial_organizations"])
+            else:
+                logger.info(
```

```suggestion
                logger.warning(
```

---

I am wondering if we want to throw an exception here?

---

It should be considered an error. Thus, we can throw a custom exception (e.g. `NoRemainingTrialsException`) with a clear, user-friendly message and prevent further processing.

---

Go for it!

---

Added!